engine/runtime/src/test/scala/org/enso/compiler/test/pass/analyse/GatherDiagnosticsTest.scala

```diff
@@ -109,6 +109,28 @@ class GatherDiagnosticsTest extends CompilerTest {
       gatheredErrros.toSet shouldEqual Set(error1, error2, error3)
     }
 
+    "work with annotations" in {
+      implicit val passManager: PassManager =
+        new Passes(defaultConfig).passManager
+
+      implicit val moduleContext: ModuleContext =
+        buildModuleContext(freshNameSupply = Some(new FreshNameSupply))
+
+      val ir =
+        """@x bar
+          |foo x = x
+          |""".stripMargin.preprocessModule
+      val result = GatherDiagnostics.runModule(ir, moduleContext)
+      val diagnostics = result
+        .unsafeGetMetadata(GatherDiagnostics, "Impossible")
+        .diagnostics
+
+      diagnostics should have size 1
+      diagnostics.map(_.message(null)) should contain theSameElementsAs Seq(
```

Comment:
Can you add a test to the [ErrorCompilerTest](https://github.com/enso-org/enso/blob/87dfb57f53a941c903cae66ceba6ce4bc4ab25a1/engine/runtime/src/test/java/org/enso/compiler/ErrorCompilerTest.java#L12) - that'd be real unit test. The test here is more integration one.